src/protocol/malicious.rs

```diff
@@ -284,30 +286,31 @@ pub mod tests {
                 b_malicious,
             );
 
-            let (ab, rab) = try_join(
-                a_ctx
-                    .narrow("SingleMult")
-                    .multiply(RecordId::from(0))
-                    .await
-                    .execute(a_shares[i], b_shares[i]),
-                a_ctx
-                    .narrow("DoubleMult")
-                    .multiply(RecordId::from(1))
-                    .await
-                    .execute(ra, b_shares[i]),
-            )
-            .await?;
+            #[allow(clippy::similar_names)]
+            let mult_result = a_ctx
+                .malicious_multiply(RecordId::from(0))
+                .await
+                .execute(a_malicious, b_malicious)
+                .await?;
 
-            acc.accumulate_macs(
-                &a_ctx.narrow(&Step::ValidateMultiplySubstep).prss(),
-                RecordId::from(0),
-                MaliciousReplicated::new(ab, rab),
-            );
+            v.validate(ctx.narrow("SecurityValidatorValidate")).await?;
 
-            v.validate(ctx.narrow("SecurityValidatorValidate")).await
+            Ok::<(MaliciousReplicated<Fp31>, Replicated<Fp31>), BoxError>((mult_result, r_share))
```

Comment:
It's super annoying that it can't figure out what kind of Error this OK should use, and thus I have to include this entire long line of garbage.

Reply:
You might be able to replace pieces of this with `_` or put an explicit return type on the closure above (like `.map(|(i, ctx)| -> Result<(MaliciousReplicated<_>, Replicated<_>), BoxError> async move {`

Reply:
OK! I can replace everything except the `BoxError` part with `_`!